### PR TITLE
Server Browser: QOL and hide server info if no server focused

### PR DESCRIPTION
**Pull request by KanjiuAkuma**
- Hide server info if no server selected (sometimes showed (the wrong server) even so no server was selected, e.g. if changing the filters)

QOL:
- Pressing enter once connects to a server (was handled by 'double click handler' before, so you had to press it twice)
- Add 'R' key bind to reload the server list (Had a controller bind, but none for keyboard)
- Auto focus password field on connect to password protected server

### CODE REVIEW

**Review by GeckoEidechse — CHANGES_REQUESTED** [review]
Tested:
- [x] Pressing `ENTER` once to connect
- [x] `R` refreshes server list
- [x] Auto-focus password field

However I noticed that with hiding the server info on no selected server it's no longer possible to use the join button in the bottom right:

https://user-images.githubusercontent.com/40122905/162637579-dc0df048-3d7c-44d0-aeed-aad21c0150e7.mp4

I'd suggest either looking if you can fix it or just dropping the hiding server info from this PR so we can get it merged faster and fixing the hide server info part in a separate one ^^

**Comment by KanjiuAkuma** [comment]
> I'd suggest either looking if you can fix it or just dropping the hiding server info from this PR so we can get it merged faster and fixing the hide server info part in a separate one ^^

Fixed it with commit [7e3c9c4](https://github.com/R2Northstar/NorthstarMods/pull/295/commits/7e3c9c42d63cbfea04ce16d7d0ef3f6c86705ef5).

Also retains focus on canceling a connect now (before all focus was lost, after pressing cancel on the connect dialog so keyboard nav wouldn't work at all at that point)

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 1]
```diff
@@ -518,6 +521,38 @@ void function OnUpArrowSelected( var button )
 	UpdateListSliderPosition( file.serversArrayFiltered.len() )
 }
 
+////////////////////////
+// Key Callbacks
+////////////////////////
+void function OnEnterPressed(arg) {
```
Comment (RoyalBlue1) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:527]: There should be a newline in front of the `{`. 

```suggestion
void function OnEnterPressed(arg) 
{
```

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 2]
```diff
@@ -518,6 +521,38 @@ void function OnUpArrowSelected( var button )
...
+	// only trigger if a server is focused
+	if (IsServerButtonFocused()) {
```
Comment (RoyalBlue1) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:529]:
```suggestion
	if (IsServerButtonFocused()) 
	{
```

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 3]
```diff
@@ -518,6 +521,38 @@ void function OnUpArrowSelected( var button )
...
+		OnServerSelected(0)
+	}
+}
+
+void function OnKeyRPressed(arg) {
```
Comment (RoyalBlue1) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:534]:
```suggestion
void function OnKeyRPressed(arg) 
{
```

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 4]
```diff
@@ -518,6 +521,38 @@ void function OnUpArrowSelected( var button )
...
+	if (!IsSearchBarFocused()) {
```
Comment (RoyalBlue1) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:535]:
```suggestion
	if (!IsSearchBarFocused()) 
	{
```

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 5]
```diff
@@ -518,6 +521,38 @@ void function OnUpArrowSelected( var button )
...
+		RefreshServers(0);
+	}
+}
+
+bool function IsServerButtonFocused() {
```
Comment (RoyalBlue1) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:540]:
```suggestion
bool function IsServerButtonFocused() 
{
```

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 6]
```diff
@@ -518,6 +521,38 @@ void function OnUpArrowSelected( var button )
...
+	var focusedElement = GetFocus();
+	var name = Hud_GetHudName(focusedElement);
+
+	foreach (element in GetElementsByClassname( file.menu, "ServerButton")) {
```
Comment (RoyalBlue1) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:544]:
```suggestion
	foreach (element in GetElementsByClassname( file.menu, "ServerButton")) 
	{
```

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 7]
```diff
@@ -518,6 +521,38 @@ void function OnUpArrowSelected( var button )
...
+		if ( element == focusedElement ) return true
```
Comment (RoyalBlue1) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:545]:
```suggestion
		if ( element == focusedElement ) 
			return true
```

**Review by GeckoEidechse — APPROVED** [review]
Looks good now.

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 8]
```diff
@@ -518,6 +521,38 @@ void function OnUpArrowSelected( var button )
...
+	}
+
+
+	return false;
+}
+
+bool function IsSearchBarFocused() {
```
Comment (RoyalBlue1) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:552]:
```suggestion
bool function IsSearchBarFocused() 
{
```

**Review by RoyalBlue1 — CHANGES_REQUESTED** [review]
Looks good otherwise

**Comment by KanjiuAkuma** [comment]
> Looks good otherwise

Ah yes... code style is different... my bad^^

**Pull request merged**